go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -1464,7 +1473,7 @@ func (tsv *TabletServer) registerHealthzHealthHandler() {
 			acl.SendError(w, err)
 			return
 		}
-		if !tsv.sm.IsServing() {
+		if tsv.IsServingType() && !tsv.sm.IsServing() {
```

Comment:
As @deepthi alluded to, this would cause tablets in a BACKUP or RESTORE state to be marked as healthy, which we don't want since a tablet in that state is not ready to serve. That's actually the reason we couldn't use `/debug/health` for the operator's ReadinessProbe and had to use `/healthz` instead, so we don't want to bring that problem to `/healthz` as well.

Reply:
Noted. I'll just explicitly check for SPARE then.

Reply:
I think that would work for what we need, and it would return `/healthz` to the Vitess 6.0 behavior with respect to SPARE. What do you think, @deepthi?

Basically we're claiming that SPARE is a special case among tablet types because it's the only time when it's normal for a tablet to stay in a non-serving state indefinitely. Things like BACKUP and RESTORE should be temporary, and you do want to get notified if a tablet stays in those states too long so it makes sense to report unhealthy.

Reply:
Done.

Reply:
This should be fine.